Lib/asyncio/selector_events.py

```diff
@@ -1101,6 +1143,16 @@ def write_eof(self):
         if not self._buffer:
             self._sock.shutdown(socket.SHUT_WR)
 
+    def writelines(self, list_of_data):
+        if self._eof:
+            raise RuntimeError('Cannot call writelines() after write_eof()')
+        if self._empty_waiter is not None:
+            raise RuntimeError('unable to writelines; sendfile is in progress')
+        if not list_of_data:
+            return
+        self._buffer.extend([memoryview(i) for i in list_of_data])
+        self._loop._add_writer(self._sock_fd, self._write_ready)
```

Comment:
The fast path is lost here.
Ideally, the code should be similar to `.write()`. Could the common part be extracted into a helper method?

Reply:
I didn't see your reply to this.

Reply:
Andrew's comment seems misplaced, there was no fastpath for writelines in asyncio, it always joined all the data in one giant bytes object followed by write. 
https://github.com/python/cpython/blob/53eef27133c1da395b3b4d7ce0ab1d5b743ffb41/Lib/asyncio/transports.py#L116-L123

With this PR if sendmsg is available then we use it as it is faster otherwise fallback to the old join and send implementation.

Reply:
Got it. Nice, actually!